### PR TITLE
Refactor Api::GetTraineesService since filter

### PR DESCRIPTION
**Pull request by Reyko**
* Don't convert to date to allow the since param to be in ISO 8601
* Change the where clause to update_at >= since

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/models/api/trainee_filter_params.rb [hunk 1]
```diff
@@ -46,5 +39,9 @@ def check_academic_cycle
         )
       end
     end
+
+    def iso8601?(string)
+      string.match?(/\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(Z|(\+|\-)\d{2}:\d{2})?\z/)
+    end
```
Comment (tomtrentham) [inline app/models/api/trainee_filter_params.rb:45]: I'm not a big fan of complex regexes like this because they are pretty hard to parse at a glance and see what's going on and if the standard were to change (however unlikely) we'd have to update this. Could we use the built-in `iso8601` method to try to parse the string?

**Review by tomtrentham — APPROVED** [review]
This looks good. Just one query about the checking of the string format.

File: app/models/api/trainee_filter_params.rb (hunk 1 above)
Comment (Reyko) [reply app/models/api/trainee_filter_params.rb:45]: Opened a [draft PR](https://github.com/DFE-Digital/register-trainee-teachers/pull/4289) to demo which cases won't be captured if we use the `DateTime.iso6801` as an example.

**Review by tomtrentham — APPROVED** [review]
🚢
